can-view-scope.js

```diff
@@ -219,99 +359,94 @@ assign(Scope.prototype, {
 					currentReads = keyReads.slice(nameIndex);
 				},
 				earlyExit: function(parentValue, nameIndex) {
-					if (nameIndex > setObserveDepth || (nameIndex === setObserveDepth && (typeof parentValue === "object" && keyReads[nameIndex].key in parentValue))) {
+					var isVariableScope = currentScope._meta.variable === true,
+						updateSetObservable = false;
+					if(isVariableScope === true && nameIndex === 0) {
+						// we MUST have pre-defined the key in a variable scope
+						updateSetObservable = canReflect.hasOwnKey( parentValue, keyReads[nameIndex].key);
+					} else {
+						updateSetObservable =
+							// Has more matches
+							nameIndex > setObserveDepth ||
+							// The same number of matches but it has the key
+							nameIndex === setObserveDepth && (typeof parentValue === "object" && canReflect.hasOwnKey( parentValue, keyReads[nameIndex].key));
```

Comment:
Will using `hasOnwKey` cause an issue with functions? I vaguely remember `keyReads[nameIndex].key in parentValue` being used in order to support functions.

Reply:
I'm not sure what you mean.  Can you explain more?  What kinds of issues?  Do you mean this should be `canReflect.hasKey()`? Yeah, might need to be `hasKey` to support values on the proto chain (which is what I think you mean by "functions").

Reply:
I just mean functions on the context so things like `{{ func() }}` will work.

I ask becuase there's this difference between using `in` and using `hasKey`:
```js
const Foo = DefineMap.extend({
  func() {}
});

const foo = new Foo();

Reflect.hasOwnKey(foo, "func"); // false
Reflect.hasKey(foo, "func"); // false
"func" in foo; // true
```

Maybe this is should be fixed in DefineMap, but that is how it works currently.

Reply:
so is this not going to be a problem @justinbmeyer ?

Reply:
I thought I changed it to hasKey.  It probably will be a problem.

Reply:
Changed it here: https://github.com/canjs/can-view-scope/commit/02a0c6c1a5a5f4a454241b97b83447618fc40307#commitcomment-30467125